python/lsst/obs/metadata/observationInfo.py

```diff
@@ -0,0 +1,170 @@
+                   "location": ("Location of the observatory", "astropy.coordinates.EarthLocation"),
+                   "exposure": ("Unique (with instrument) integer identifier for this Exposure", "int"),
+                   "visit": ("""ID of the Visit this Exposure is associated with.
+
+    Science observations should essentially always be
+    associated with a visit, but calibration observations
+    may not be.""", "int"),
+                   "abstract_filter": ("Generic name of this filter", "str"),
+                   "physical_filter": ("The bandpass filter used for all exposures in this Visit.", "str"),
+                   "datetime_begin": ("Timestamp of the start of the Exposure.", "astropy.time.Time"),
+                   "datetime_end": ("Timestamp of the end of the Exposure.", "astropy.time.Time"),
+                   "exposure_time": ("Duration of the Exposure with shutter open (seconds).", "float"),
+                   "dark_time": ("Duration of the Exposure with shutter closed (seconds).", "float"),
+                   "boresight_airmass": ("Airmass of the boresight of the telescope", "float"),
+                   "boresight_rotation_angle": ("Angle of the instrument in boresight_rotation_coord frame",
+                                                "astropy.coordinates.Angle"),
+                   "boresight_rotation_coord": ("Coordinate frame of the instrument rotation angle"
+                                                " (sky, unknown)", "str"),
+                   "detector_num": ("Unique (for instrument) integer identifier for the sensor", "int"),
+                   "detector_name": ("Name of the detector within the instrument (might not be unique)",
+                                     "str"),
+                   "detector_exposure_id": ("Integer identifier for this detector in this exposure",
+                                            "int"),
+                   "object": ("Object of interest or field name", "str"),
+                   "temperature": ("Temperature outside the dome", "astropy.units.Quantity"),
+                   "pressure": ("Atmospheric pressure outside the dome", "astropy.units.Quantity"),
+                   "relative_humidity": ("Relative humidity outside the dome", "float"),
+                   "tracking_radec": ("Requested RA/Dec to track", "astropy.coordinates.SkyCoord"),
+                   "altaz_begin": ("Telescope boresight azimuth and elevation at start of observation",
+                                   "astropy.coordinates.AltAz"),
+                   "science_program": ("Observing program (survey or proposal) identifier", "str"),
+                   "obstype": ("Type of observation (science, dark, flat, bias, focus etc)", "str"),
+                   "obsid": ("Unique observation identifier", "str")}
+
+    """All the properties supported by this class with associated
+    documentation."""
+
+    def __init__(self, header, translator_class=None):
```

Comment:
should keyword arguments be upper cased?

Reply:
@TallJimbo or @ktlim do you want to jump in here?

Reply:
Since types are objects, I think any attempt to follow this question through to any logical conclusion will end in sadness.  In terms of existing practice I think it's most common to use uppercase for class attributes and lowercase for keyword arguments, while we collectively haven't developed an implicit preference for local variables.